diff --git a/dist/internal/helpers/overridable.js b/dist/internal/helpers/overridable.js
index 4bfd74ec7fc838dbce59491c37180f9a3cd798c2..a92e724ef14e9148e5bdf40c1e5a1e5817d61ed7 100644
--- a/dist/internal/helpers/overridable.js
+++ b/dist/internal/helpers/overridable.js
@@ -1,10 +1,10 @@
-export const overridable = (store, onChange) => {
+export const overridable = async (store, onChange) => {
     const update = (updater, sideEffect) => {
-        store.update((curr) => {
+        store.update(async (curr) => {
             const next = updater(curr);
             let res = next;
             if (onChange) {
-                res = onChange({ curr, next });
+                res = await onChange({ curr, next });
             }
             sideEffect?.(res);
             return res;